Library/Homebrew/formula_installer.rb

```diff
@@ -146,11 +146,16 @@ def check_install_sanity
     raise FormulaInstallationAlreadyAttemptedError, formula if @@attempted.include?(formula)
 
     unless skip_deps_check?
-      unlinked_deps = formula.recursive_dependencies.map(&:to_formula).select do |dep|
+      recursive_deps = formula.recursive_dependencies.map(&:to_formula)
+      unlinked_deps = recursive_deps.select do |dep|
         dep.installed? && !dep.keg_only? && !dep.linked_keg.directory?
       end
       raise CannotInstallFormulaError,
         "You must `brew link #{unlinked_deps*" "}` before #{formula.full_name} can be installed" unless unlinked_deps.empty?
+
+      pinned_deps = recursive_deps.select(&:pinned?)
+      raise CannotInstallFormulaError,
+        "You must `brew unpin #{pinned_deps*" "}` before #{formula.full_name} can be installed" unless pinned_deps.empty?
```

Comment:
May be worth clarifying a little bit `as installing formula.full_name requires the latest version of pinned_deps`